File: clean-it-frontend/cleanit/lib/models/pickup.dart
import 'package:get/get_connect/http/src/request/request.dart';
import 'package:hive/hive.dart';

part 'pickup.g.dart';

@HiveType(typeId: 5)
class PickupList extends HiveObject {
  @HiveField(0)
  List<PickUp> pickups;
  PickupList({required this.pickups});
}

@HiveType(typeId: 3)
class PickUp extends HiveObject {
  @HiveField(0)
  String email;
  @HiveField(1)
  String name;
  @HiveField(2)
  String address1;
  @HiveField(3)
  String pincode;
  @HiveField(4)
  String state;
  @HiveField(5)
  String city;
  @HiveField(6)
  String pickupdate;
  @HiveField(7)
  String pickuptime;
  @HiveField(8)
  List<PickUpItems> items;
  @HiveField(9)
  String price;
  @HiveField(10)
  List<Status> status;
  @HiveField(11)
  String phone;

  PickUp(
      {required this.email,
      required this.name,
      required this.address1,
      required this.city,
      required this.phone,
      required this.items,
      required this.pickupdate,
      required this.pickuptime,
      required this.pincode,
      required this.price,
      required this.state,
      required this.status});
}

@HiveType(typeId: 4)
class PickUpItems extends HiveObject {
  @HiveField(0)
  String name;
  @HiveField(1)
  String quantity;
  @HiveField(2)
  String type;

  PickUpItems({required this.name, required this.quantity, required this.type});
}

@HiveType(typeId: 6)
class Status extends HiveObject {
  @HiveField(0)
  String date;
  @HiveField(1)
  String time;

  Status({required this.date, required this.time});
}

File: clean-it-backend/models/Pickup.js
const mongoose = require('mongoose');
const Joi = require('joi');

const PickupSchema = mongoose.Schema({
    email : {type: String, min: 5, max: 255, required: true},
    name : {type:String, min: 1, max: 255, required: true},
    price : {type:String, min: 1, max: 255, required: true},
    address1 : {type:String, min: 1, max: 255, required: true},
    phone : {type: String, min: 10, max: 10, required: true},
    pincode : {type:String, min: 6, max: 6, required: true},
    city : {type:String, min: 1, max: 255, required: true},
    state : {type:String, min: 1, max: 255, required: true},
    items : {type: [], required: true},
    status : {type: [], required: true},
    pickupdate : {type: String, required: true},
    pickuptime : {type: String, required: true}
});

const Pickup = new mongoose.model("Pickup", PickupSchema);

function validatePickup(pickup){
    const schema = {
        email : Joi.string().email().required(),
        name : Joi.string().min(1).max(255).required(),
        price : Joi.string().min(1).max(255).required(),
        address1 : Joi.string().min(1).max(255).required(),
        phone : Joi.string().min(10).max(10).required(),
        pincode : Joi.string().min(6).max(6).required(),
        city : Joi.string().min(1).max(255).required(),
        state : Joi.string().min(1).max(255).required(),
        items : Joi.array().required(),
        status : Joi.array().required(),
        pickupdate : Joi.string().min(1).max(255).required(),
        pickuptime : Joi.string().min(1).max(255).required(),
    }

    return Joi.validate(pickup, schema);
}


module.exports.Pickup = Pickup
module.exports.validatePickup = validatePickup